EIPS/eip-7723.md

```diff
@@ -0,0 +1,80 @@
+
+When the Upgrade Meta EIP is moved to `Last Call`, the `Proposed for Inclusion` and `Considered for Inclusion` lists **SHOULD** be removed, leaving only `Scheduled for Inclusion`. 
+
+Before the Upgrade Meta EIP is moved to `Final`, the `Scheduled for Inclusion` stage **MUST** be renamed to `Included` and contain only EIPs that were activated in the upgrade. 
+
+### Proposed for Inclusion
+
+To formally propose the inclusion of a Core EIP in a network upgrade, someone **MUST** open a pull request against the Upgrade Meta EIP to add the EIP in the `Proposed for Inclusion` section. Reasonable pull requests **SHOULD** be merged in a timely fashion by the Upgrade Meta EIP author.  
+
+At this stage, implementation teams **SHOULD** review the EIP in the context of including it in the next upgrade. 
+
+Note that EIPs must be `Proposed for Inclusion` for each network upgrade. In other words, proposals do not "carry over" to the next upgrade if an EIP is not included in the one it was first proposed for. 
+
+### Considered for Inclusion 
+
+Once client developers have reviewed an EIP which was `Proposed for Inclusion`, they **MAY** move it to the `Considered for Inclusion` stage. Once a decision is made by client teams to move an EIP to `Considered for Inclusion`, the Upgrade Meta EIP **MUST** be updated to reflect this.
+
+`Considered for Inclusion` signals that client developers are generally positive towards the EIP, and that, assuming it meets all the requirements for mainnet deployment, it **MAY** be included in the network upgrade. This stage is similar to "concept ACK" in other open source projects, and is not sufficient to result in deployment to mainnet. An EIP **MAY** be removed from `Considered for Inclusion` if client teams are generally against including the EIP in the network upgrade. 
+
+### Scheduled for Inclusion 
+
+When client developers decide to work on an EIP as part of a network upgrade, the EIP **SHOULD** move to the `Scheduled for Inclusion` stage, and the Upgrade Meta EIP **MUST** be updated to reflect this. 
+
+`Scheduled for Inclusion` signals that implementation and testing work are underway, and that, assuming no issues arise as part of this process, the EIP **SHOULD** be included in the network upgrade. An EIP **MAY** be removed from `Scheduled for Inclusion` if client teams are generally against including the EIP in the network upgrade. 
+
+### Included
+
+Once a network upgrade has been activated, all EIPs that were part in the upgrade **MUST** be moved to `Included` in the Upgrade Meta EIP, and the `Proposed for Inclusion`, `Considered for Inclusion` and `Scheduled for Inclusion` lists **MUST** be removed from the Meta EIP. 
```

Comment:
I am not sure we need 3 states. Status `CFI` and `SFI` sound confusing. If an EIP is `CFI` should clients implement it already or not? If not, what's the point of this status? Also, historically clients have implemented EIPs at different times depending on their priorities, regardless of overall consensus.

I assume all EIPs in existence are meant to be included eventually, so all EIPs are automatically `Proposed for Inclusion`. Following the proposed work-flow of doing PRs against the meta-EIP, the existence of this PR already shows intent for the EIP to be scheduled.

Why not have a single list: `Scheduled for Inclusion`? Consider the following flow:

- EIP is drafted
- EIP champion opens PR to the meta-EIP (= shows intent to be scheduled)
- Client developers, community members, and affected stakeholders can signal positive intent on the PR / forum
- Once rough consensus forms for inclusion, the PR is merged. At this point the EIP is expected to be implemented by clients

Reply:
@dapplion I replied to this on [EthMag](https://ethereum-magicians.org/t/eip-7723-network-upgrade-inclusion-stages/20281/5?u=timbeiko)